### PR TITLE
[quant] Add quantized Sigmoid module

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45920 [quant][fx][graphmode][api] Change API for custom module
* #45919 [quant][fx][graphmode] Run symbolic_trace in quantization
* #45885 [quant][fx][graphmode] Add support for sigmoid
* #45884 [quant][eagermode] Add support for sigmoid module
* **#45883 [quant] Add quantized Sigmoid module**
* #45882 [quant] Add quantized::sigmoid that take output_scale/output_zero_point as input

Summary:

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_sigmoid

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24129116](https://our.internmc.facebook.com/intern/diff/D24129116)

### CODE REVIEW

File: torch/nn/quantized/modules/activation.py [hunk 1]
```diff
@@ -149,3 +149,24 @@ def _get_name(self):
...
+
+class Sigmoid(torch.nn.Sigmoid):
+    r"""This is the quantized equivalent of :class:`~torch.nn.LeakyReLU`.
+
+    Args:
+        scale: quantization scale of the output tensor
+        zero_point: quantization zero point of the output tensor
+    """
+
+    def __init__(self, output_scale: float, output_zero_point: int):
```
Comment (z-a-f) [inline torch/nn/quantized/modules/activation.py:161]: Can we have default args? It is safe to assume those for this function

**Review by z-a-f — APPROVED** [review]

File: torch/nn/quantized/modules/activation.py (hunk 1 above)
Comment (jerryzh168) [reply torch/nn/quantized/modules/activation.py:161]: not sure, it's not likely that user is going to construct quantized model from quantized modules from scratch anyways, what is the benefit of having default args?

File: torch/nn/quantized/modules/activation.py [hunk 2]
```diff
@@ -149,3 +149,24 @@ def _get_name(self):
     def from_float(cls, mod):
         scale, zero_point = mod.activation_post_process.calculate_qparams()
         return cls(float(scale), int(zero_point), mod.negative_slope, mod.inplace)
+
+class Sigmoid(torch.nn.Sigmoid):
+    r"""This is the quantized equivalent of :class:`~torch.nn.LeakyReLU`.
```
Comment (raghuramank100) [inline torch/nn/quantized/modules/activation.py:154]: The docs are off here, can we refer to sigmoid instead of leakyRelu?

Comment (jerryzh168) [reply torch/nn/quantized/modules/activation.py:154]: oh sorry, will fix in next PR
